base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1985,11 +2023,19 @@ fn reorganize_chain<T: BlockchainBackend>(
             restore_reorged_chain(backend, fork_hash, removed_blocks)?;
             return Err(e.into());
         }
-
-        insert_best_block(&mut txn, block.clone())?;
+        txn.delete_orphan(block_hash);
         // Failed to store the block - this should typically never happen unless there is a bug in the validator
         // (e.g. does not catch a double spend). In any case, we still need to restore the chain to a
         // good state before returning.
+        if let Err(e) = insert_best_block(&mut txn, block.clone()) {
+            warn!(
+                target: LOG_TARGET,
+                "Failed extract Monero data: {:?}. Restoring last chain.", e
```

Comment:
This doesn't look like the right error. Is it correct?

Reply:
Yeah it is correct. 
Insert best block can only fail if it fails to deserialise the monero data.

Reply:
nit: That is an internal detail of insert_best_block though, so the warn log text should be agnostic to that. The `e` can/should describe the problem itself.